### ISSUE TITLE
[HOLD for payment 2024-06-28] [$250] [Search v1] Live search updates

### ISSUE THREAD

**Issue by luacmartins** [Expensify/App#41626]
Section from design doc - https://docs.google.com/document/d/1ldZ85fyQO4662hLZRNFfm0_IxaP7-PWvaqc8Z33jsLQ/edit#bookmark=kix.79jwwhj0fjyz

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01d2ae989d15bcd2d3</li>
        <li>Upwork Job ID: 1790718970882084864</li>
        <li>Last Price Increase: 2024-05-15</li>
<li>Automatic offers: </li>
<ul>
<li>rayane-djouah | Reviewer | 0</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @allroundexperts</details>

**Comment by luacmartins** [comment]
Taking off hold, we'll start implementation shortly.

**Comment by Kicu** [comment]
@luacmartins 
I looked at this issue and it feels like we need some working api in order to start working on it? Any ETA on when backend changes will be done, or can we perhaps work on this already?

**Comment by luacmartins** [comment]
No API needed for this one. We only need to update Onyx so that any incoming update is processed in a way that it also updates the data in the `snapshot_` key.

**Comment by luacmartins** [comment]
@Kicu I assigned you to this issue. Let me know if you can't work on it.

**Comment by Kicu** [comment]
Will work on it 👍

**Comment by Kicu** [comment]
@luacmartins do you mind checking my comment here? https://docs.google.com/document/d/1ldZ85fyQO4662hLZRNFfm0_IxaP7-PWvaqc8Z33jsLQ/edit?disco=AAABJtWc-70

**Comment by Christinadobrzyn** [comment]
looks like @luacmartins is commenting in the d[esign doc](https://docs.google.com/document/d/1ldZ85fyQO4662hLZRNFfm0_IxaP7-PWvaqc8Z33jsLQ/edit#heading=h.tipze2z8hbad)

**Comment by Christinadobrzyn** [comment]
looks like @luacmartins and @Kicu are working through the comments in the [design doc](https://docs.google.com/document/d/1ldZ85fyQO4662hLZRNFfm0_IxaP7-PWvaqc8Z33jsLQ/edit#heading=h.tipze2z8hbad)

**Comment by luacmartins** [comment]
Looking for volunteers

**Comment by trjExpensify** [comment]
@Kicu is working on this one, though Mateusz, let me know if that's not correct!

**Comment by Kicu** [comment]
No Im not actively Im sorry for the confusion, This is not being worked on and free to take.

At some point there were multiple tasks for Search to take and I focused on other ones, please unassign me for now at least

**Comment by trjExpensify** [comment]
Oh okay, I've asked here to see if Jan can take it instead then: https://expensify.slack.com/archives/C04878MDF34/p1715787635557759?thread_ts=1715684435.196609&cid=C04878MDF34

**Comment by jnowakow** [comment]
I'm from Software Mansion and I'll work on it

**Comment by trjExpensify** [comment]
Thanks!

**Comment by Christinadobrzyn** [comment]
Looks like we're working on this, @jnowakow let us know if you need anything or have a specific update!

**Comment by Christinadobrzyn** [comment]
Thanks for the heads up @rayane-djouah! For a MelvBot update - the PR is in the works!

**Comment by Christinadobrzyn** [comment]
Melvbot update: @jnowakow is working on the PR - https://github.com/Expensify/react-native-onyx/pull/552

**Comment by luacmartins** [comment]
Thanks for letting me know @rayane-djouah! Reassigned to @allroundexperts

**Comment by Christinadobrzyn** [comment]
Update for MelvinBot - PR is in the works - https://github.com/Expensify/react-native-onyx/pull/552

I think we can move this to weekly while that's in progress, do you think?

**Comment by luacmartins** [comment]
Yea, this is in review

**Comment by Christinadobrzyn** [comment]
monitoring - https://github.com/Expensify/react-native-onyx/pull/552

**Comment by luacmartins** [comment]
That PR is on hold for merge until we fix the Onyx performance regression on the existing and latest Onyx version

**Comment by Christinadobrzyn** [comment]
Adding [Hold for #42772] in the subject.

**Comment by Christinadobrzyn** [comment]
Just a heads up - I'm going to be ooo until June 24th but I don't think this needs to be reassigned. If it does, please feel free to do so while I'm away.

**Comment by luacmartins** [comment]
@jnowakow we merged the Onyx PR, can you prepare a PR to bump the version in App too?

**Comment by jnowakow** [comment]
@luacmartins sure, I'll do it

**Comment by jnowakow** [comment]
@luacmartins [PR is ready](https://github.com/Expensify/App/pull/43764).

**Comment by Christinadobrzyn** [comment]
Hum... the upwork contract has an offer sent to Rayane D. 

Are we paying out anyone other than @allroundexperts and @jnowakow?

**Comment by jnowakow** [comment]
@Christinadobrzyn I'm from SoftwareMansion, an expert agency, so I'm not receiving direct payments 😅

**Comment by Christinadobrzyn** [comment]
Thanks @jnowakow! 

[This explains](https://github.com/Expensify/App/issues/41626#issuecomment-2127836521) why we sent an offer to @rayane-djouah so I'm going to withdraw that offer. 

Payment summary here - https://github.com/Expensify/App/issues/41626#issuecomment-2197411154

@allroundexperts can you let me know about a regression test?

**Comment by Christinadobrzyn** [comment]
moving this to weekly since all we're waiting on is a regression test, if applicable.

**Comment by luacmartins** [comment]
No regression tests needed. We'll add them as part of the project wrap up. We can close this then.

**Comment by JmillsExpensify** [comment]
$250 approved for @allroundexperts
